tests/ondemand/ondemand_number_in_string_tests.cpp

```diff
@@ -0,0 +1,122 @@
+#include "simdjson.h"
+#include "test_ondemand.h"
+#include <string>
+
```

Comment:
Is the new functionality compatible with the JSON Pointer approach, if so we should add a few tests.

Reply:
(If it is not compatible with JSON Pointer, then we should investigate.)